### PR TITLE
feat!: Rename title props for Alert, Header and Dialog and require it for Dialog

### PR DESCRIPTION
**Pull request by dlnr**
Dialog: title -> heading (required)
Alert: title -> heading (optional)
Header: title -> siteName

And some missing tests

### CODE REVIEW

**Review by VincentSmedinga — DISMISSED** [review]

File: storybook/src/components/Header/Header.docs.mdx [hunk 1]
```diff
@@ -32,14 +32,14 @@ A Page Menu in the Header should not wrap.
 
 <Canvas of={HeaderStories.WithLinks} />
 
-## With links and menu
+## With app name and menu
```
Comment (alimpens) [inline storybook/src/components/Header/Header.docs.mdx:35]: This doesn't have an app name?

Comment (dlnr) [reply storybook/src/components/Header/Header.docs.mdx:35]: I blame Vincent use of webstorm ;)

Comment (VincentSmedinga) [reply storybook/src/components/Header/Header.docs.mdx]: Unjustly so, you know I’m an excellent renamer 😎

It was actually a slip of your own keyboard: https://github.com/Amsterdam/design-system/commit/6e86af8c117cee786afd1896e35db79d19c1085b#diff-14fc32124a34d0ee7bc7ac3376a0aab2702a9e7dc359307662fab46821ddf114R35

Comment (dlnr) [reply storybook/src/components/Header/Header.docs.mdx]: 🤦‍♂️

**Review by VincentSmedinga — APPROVED** [review]

**Pull request merged**